cmd/background-newdisks-heal-ops.go

```diff
@@ -524,10 +545,10 @@ func monitorLocalDisksAndHeal(ctx context.Context, z *erasureServerPools) {
 					if err := healFreshDisk(ctx, z, disk); err != nil {
 						globalBackgroundHealState.setDiskHealingStatus(disk, false)
 						timedout := OperationTimedOut{}
-						if !errors.Is(err, context.Canceled) && !errors.As(err, &timedout) {
+						if !errors.Is(err, context.Canceled) && !errors.As(err, &timedout) && !errors.Is(err, errRetryHealing) {
 							printEndpointError(disk, err, false)
+							return
```

Comment:
I should not return here, otherwise, a disk that returned errRetryHealing will not be healed again because of globalBackgroundHealState..popHealLocalDisks

Reply:
Oh yes it must be outside